pkg/rotate/rotatekeys.go

```diff
@@ -246,7 +246,7 @@ func updateKeyLocation(account string, keyLocations config.KeyLocations, keyID,
 
 		var updated location.UpdatedLocation
 
-		if updated, err = locationToUpdate.Write(keyLocations.ServiceAccountName, keyID, key, creds); err != nil {
+		if updated, err = locationToUpdate.Write(keyLocations.ServiceAccountName, keyID, key, keyProvider, creds); err != nil {
```

Comment:
This is tricky, as I think it's only GitHub that needs the keyProvider in order to decide whether to do a base-64 encoding.  However I think it's probably necessary as otherwise rotate would have to know about GitHub, or we'd have to add an extra function call to the loop for encoding.